apps/client/app/components/PurchaseOrderForm.tsx

```diff
@@ -0,0 +1,167 @@
+// This is the only client component so far in the application due to the need for interactivity
+'use client'
+
+import React, { useEffect, useState } from 'react';
+import { useForm, Controller, useFieldArray } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { PurchaseOrder, Item } from '../interfaces';
+
+// This component is passed in the Purchase Order and whether we are creating or editing
+interface PurchaseOrderFormProps {
+  purchaseOrder?: PurchaseOrder;
+  mode: 'create' | 'edit';
+}
+
+// Define the validation schema using Yup
+const validationSchema = Yup.object().shape({
```

Comment:
Using Yup for client-side validation. There is an opportunity to centralize the validation code and use it on the server side as well using ValidationPipes. Seems like overkill though since we are using decorators on the DTO for server-side validation and not sure if we want to remove that.

Reply:
I did consider Zod as well and based on my research, both Zod and Yup were very similar and I just happened to choose Yup. In terms of sharing the server-side validation, it didn't occur to me to share the DTO validation to the front-end but that makes a whole lot of sense and is a better approach than sharing the client-side validation to the back-end.